src/code/z_player_lib.c

```diff
@@ -1,153 +1,1021 @@
+}
 
 #pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_80123810.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_80123960.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_801239AC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_80123AA4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_80123BD4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_80123C58.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/Player_SetEquipmentData.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_80123D50.s")
+extern u8 sActionModelGroups[];
+
+s32 Player_ActionToModelGroup(Player* player, s32 actionParam) {
+    s32 modelGroup = sActionModelGroups[actionParam];
+
+    if ((modelGroup == 2) && func_801234B0(player)) {
+        return 1;
+    }
+    return modelGroup;
+}
+
+extern u8 D_801BFF90[PLAYER_FORM_MAX];
+#if 0
+u8 D_801BFF90[PLAYER_FORM_MAX] = {
+    PLAYER_BOOTS_FIERCE_DEITY, // PLAYER_FORM_FIERCE_DEITY
+    PLAYER_BOOTS_GORON,        // PLAYER_FORM_GORON
+    PLAYER_BOOTS_ZORA_LAND,    // PLAYER_FORM_ZORA
+    PLAYER_BOOTS_DEKU,         // PLAYER_FORM_DEKU
+    PLAYER_BOOTS_HYLIAN,       // PLAYER_FORM_HUMAN
+};
+#endif
+
+extern u8 D_801BFF98[PLAYER_FORM_MAX];
+#if 0
+u8 D_801BFF98[PLAYER_FORM_MAX] = {
+    1, // PLAYER_FORM_FIERCE_DEITY
+    3, // PLAYER_FORM_GORON
+    2, // PLAYER_FORM_ZORA
+    0, // PLAYER_FORM_DEKU
+    1, // PLAYER_FORM_HUMAN
+};
+#endif
+
+typedef struct {
+    /* 0x00 */ u8 flag;
+    /* 0x02 */ u16 textId;
+} TextTriggerEntry; // size = 0x04
+
+extern TextTriggerEntry D_801BFFA0[];
+#if 0
+TextTriggerEntry D_801BFFA0[] = {
+    { 1, 0x26FC },
+    { 2, 0x26FD },
+    { 0, 0 },
+    { 2, 0x26FD },
+};
+#endif
+
+extern Gfx** sPlayerDListGroups[];
+
+void func_801239AC(Player* player) {
+    if (player->stateFlags1 & PLAYER_STATE1_400000) {
+        if ((player->heldItemActionParam < 0) || (player->heldItemActionParam == player->itemActionParam)) {
+            if (!Player_HoldsTwoHandedWeapon(player)) {
+                if (!func_801234B0(player)) {
+                    D_801F59E0 = player->transformation * 2;
+                    player->rightHandType = 8;
+                    player->rightHandDLists = &sPlayerDListGroups[8][D_801F59E0];
+
+                    if (player->sheathType == 14) {
+                        player->sheathType = 12;
+                    } else if (player->sheathType == 15) {
+                        player->sheathType = 13;
+                    }
+
+                    player->sheathDLists = &sPlayerDListGroups[player->sheathType][D_801F59E0];
+                    player->modelAnimType = 2;
+                    player->heldItemActionParam = -1;
+                }
+            }
+        }
+    }
+}
+
+void Player_SetModels(Player* player, s32 modelGroup) {
+    u8* aux;
+
+    D_801F59E0 = player->transformation * 2;
```

Comment:
This almost makes me wonder if these are a 3d array... didn't OoT have a PR changing these recently?

Reply:
Are you referring to this PR? 
https://github.com/zeldaret/oot/pull/1143/files#diff-6352b604d32ba625e89c08478eeaadb8719fe0240427e998ccd7b98c425b2d92

That array changed in player actor, not in player lib

Reply:
Probably. But the `* 2` Just Looks Weird.

Reply:
It is probably because of all this dlists
https://github.com/Decompollaborate/mm/blob/3f7ff526991af1b5e67b0457ca9f8c8e35808253/src/code/z_player_lib.c#L744-L808